src/streaming/text/TextTracks.js

```diff
@@ -557,35 +566,50 @@ function TextTracks(config) {
         }
     }
 
+    function _handleCaptionEvents(currentItem, timeOffset) {
+        let cue = _getCueInformation(currentItem, timeOffset)
+
+        cue.onenter = function () {
+            // HTML Tracks don't trigger the onexit event when a new cue is entered,
+            // we need to manually trigger it
+            if (currentItem.type == 'html' && currentCaptionEventCue && currentCaptionEventCue.cueID !== cue.cueID) {
```

Comment:
Please use strong equality here, also see comment above for adding a separate function for this check